File: Project/java/src/main/java/com/basanta/todo/Category.java
package com.basanta.todo;

import java.util.List;

/**
 * Task category constants (language-specific: immutable list).
 * Extensible: add new values and include in ALL_CATEGORIES.
 */
public final class Category {
    public static final String WORK = "WORK";
    public static final String PERSONAL = "PERSONAL";
    public static final String SHOPPING = "SHOPPING";
    public static final String HEALTH = "HEALTH";
    public static final String OTHER = "OTHER";

    /** Immutable list of all valid categories. */
    public static final List<String> ALL_CATEGORIES = List.of(WORK, PERSONAL, SHOPPING, HEALTH, OTHER);

    private Category() {}

    public static boolean isValidCategory(String value) {
        return value != null && ALL_CATEGORIES.contains(value);
    }
}

File: Project/javaScript/src/Category.js
/**
 * Task category constants. Extensible: add new keys and include in ALL_CATEGORIES.
 * @readonly
 */
export const WORK = 'WORK';
export const PERSONAL = 'PERSONAL';
export const SHOPPING = 'SHOPPING';
export const HEALTH = 'HEALTH';
export const OTHER = 'OTHER';

/** @type {readonly string[]} */
export const ALL_CATEGORIES = Object.freeze([
  WORK,
  PERSONAL,
  SHOPPING,
  HEALTH,
  OTHER,
]);

/**
 * @param {string} value
 * @returns {boolean}
 */
export function isValidCategory(value) {
  return typeof value === 'string' && ALL_CATEGORIES.includes(value);
}

export default {
  WORK,
  PERSONAL,
  SHOPPING,
  HEALTH,
  OTHER,
  ALL_CATEGORIES,
  isValidCategory,
};
